es6-lib/services/spatial.js

```diff
@@ -197,9 +197,10 @@ class SpatialService {
             core.bufferResponse((bufferedResponse) => {
               req.log.info(`Finished upsert response ${upsertResponse.statusCode}`);
               if (bufferedResponse.statusCode > 300) {
+                req.log.error(`Core returned upsert error: ${bufferedResponse}`);
                 return cb({
                   statusCode: 503,
-                  reason: bufferedResponse.body,
+                  reason: "Upstream Error",
```

Comment:
They are JSON errors, but there's no consistency. That coupled with the fact that frontend doesn't handle JSON errors, means the only thing to do at this point is to return a generic error for internal failures and log the actual content. :(